src/VisualStudio/Core/Def/ProjectSystem/MiscellaneousFilesWorkspace.cs

```diff
@@ -333,4 +333,49 @@ protected override void ApplyDocumentTextChanged(DocumentId documentId, SourceTe
             }
         }
     }
+
+    private bool TryOpenDocumentInMetadataWorkspace(string filePath, SourceTextContainer sourceTextContainer)
+    {
+        var workspace = _fileTrackingMetadataAsSourceService.Value.TryGetWorkspace();
+        if (workspace is null)
+        {
+            // If we haven't even created a MetadataAsSource workspace yet, then this file definitely cannot be added to
+            // it. This happens when the MiscWorkspace calls in to just see if it can attach this document to the
+            // MetadataAsSource instead of itself.
+            return false;
+        }
```

Comment:
interesting.  i suppose this is a fine optimization.  but i'd love simplicity if we always had these types alive, and didn't necessarily need these checks.

Reply:
Agree, but a couple of notes
1.  Matches the previous behavior for VS, and I think changing that is a bit out of scope
2.  We're a bit better now in LSP as this kind of check no longer happens there.